pkg/github/github.go

```diff
@@ -9,11 +9,37 @@ import (
 	"net/url"
 	"strings"
 
+	"github.com/bradleyfalzon/ghinstallation/v2"
+
 	"github.com/google/go-github/v53/github"
 	"golang.org/x/crypto/nacl/box"
 	"golang.org/x/oauth2"
 )
 
+type Config struct {
+	GithubURL            string
+	GithubToken          string
+	GithubPrivateKeyFile string
+	GithubAppID          int64
+}
+
+func (c *Config) Enabled() bool {
+	return c.GithubToken != "" || (c.GithubAppID != 0 && c.GithubPrivateKeyFile != "")
+}
+
+func newGithubClient(cfg Config, owner, repo string) (*github.Client, error) {
+	if cfg.GithubToken != "" {
+		return newGithubClientWithToken(cfg.GithubToken, cfg.GithubURL)
+	} else if cfg.GithubAppID != 0 && cfg.GithubPrivateKeyFile != "" {
+		installationID, err := getInstallationID(cfg, owner, repo)
+		if err != nil {
+			return nil, err
+		}
+		return newGithubClientWithApp(cfg.GithubPrivateKeyFile, cfg.GithubAppID, installationID, cfg.GithubURL)
+	}
+	return nil, fmt.Errorf("insufficient information provided. Github client can't be created")
```

Comment:
nit: Unnecessary else statement. Could either do:

```suggestion
	if cfg.GithubToken != "" {
		return newGithubClientWithToken(cfg.GithubToken, cfg.GithubURL)
	}
	if cfg.GithubAppID != 0 && cfg.GithubPrivateKeyFile != "" {
		installationID, err := getInstallationID(cfg, owner, repo)
		if err != nil {
			return nil, err
		}
		return newGithubClientWithApp(cfg.GithubPrivateKeyFile, cfg.GithubAppID, installationID, cfg.GithubURL)
	}
	return nil, fmt.Errorf("insufficient information provided. Github client can't be created")
```

Or since this is selecting between multiple things to decide what to return, a switch statement could be appropriate and improve readability:

```suggestion
	switch {
	case cfg.GithubToken != "":
		return newGithubClientWithToken(cfg.GithubToken, cfg.GithubURL)
	case cfg.GithubAppID != 0 && cfg.GithubPrivateKeyFile != "":
		installationID, err := getInstallationID(cfg, owner, repo)
		if err != nil {
			return nil, err
		}
		return newGithubClientWithApp(cfg.GithubPrivateKeyFile, cfg.GithubAppID, installationID, cfg.GithubURL)
	default:
		return nil, fmt.Errorf("insufficient information provided. Github client can't be created")
	}
```

Reply:
done [6161e52](https://github.com/jadolg/klum/pull/25/commits/6161e5268891d9ad95c11f3a30bcd90323f357b2)